### PR TITLE
Call auotoconf, probable cause of no install-sh

### PR DESCRIPTION
**Pull request by genuinelucifer**
Just trying to test if it would work

### CODE REVIEW

**Comment by cbm755** [comment]
:-)

This is how I debug Travis CI too: slowly, one change at a time.  Can always squash them all later!

Some hints:

* you can temporarily delete the two non-pytave builds if you want (e.g., to save some electricity).

* Within the Travis CI interface, I can cancel builds: round button like `(x)`.  Not sure if you will be allowed too or not.

**Comment by genuinelucifer** [comment]
> :-)
>
> This is how I debug Travis CI too: slowly, one change at a time. Can
always squash them all later!

:)

> Some hints:
>
> you can temporarily delete the two non-pytave builds if you want (e.g.,
to save some electricity).

Hmm. Definitely.

> Within the Travis CI interface, I can cancel builds: round button like
(x). Not sure if you will be allowed too or not.

Well, I'm not getting the button. But, I guess the build is taking less
than 5 mins which is okay.

Thanks for the hints :)

**Comment by genuinelucifer** [comment]
Now, the build goes from configure to 'make' stage. Then gives following (octave-related) error:
```
In file included from /usr/include/octave-4.0.0/octave/dim-vector.h:35:0,

                 from /usr/include/octave-4.0.0/octave/Array.h:35,

                 from /usr/include/octave-4.0.0/octave/boolMatrix.h:27,

                 from /usr/include/octave-4.0.0/octave/mx-base.h:32,

                 from /usr/include/octave-4.0.0/octave/Matrix.h:30,

                 from /usr/include/octave-4.0.0/octave/oct.h:33,

                 from pytave.cc:32:

/usr/include/octave-4.0.0/octave/oct-refcount.h:27:3: error: #error "The file <octave/config.h> must be included before oct-refcount.h."

 # error "The file <octave/config.h> must be included before oct-refcount.h."

   ^

In file included from /usr/include/octave-4.0.0/octave/mx-base.h:28:0,

                 from /usr/include/octave-4.0.0/octave/Matrix.h:30,

                 from /usr/include/octave-4.0.0/octave/oct.h:33,

                 from pytave.cc:32:

/usr/include/octave-4.0.0/octave/MatrixType.h:36:1: error: variable ‘OCTAVE_API MatrixType’ has initializer but incomplete type

 MatrixType

 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:38:1: error: expected primary-expression before ‘public’
```

**Comment by cbm755** [comment]
Oh, I forget that PyTave may not build on Octave 4...  you know more about that than me: can you built pytave master against @mtmiller's PPA Octave on your own machine?  (I don't run Ubuntu but IIRC you were trying this sort of thing a few weeks back).

**Comment by genuinelucifer** [comment]
oh, i too forgot that.
yes. I'll try that. But how do i install mtmiller's PPA octave?

**Comment by cbm755** [comment]
See the .travis file, it uses it

**Comment by genuinelucifer** [comment]
oh, thanks. I'll try and post the results.

**Comment by genuinelucifer** [comment]
I tried the ppa but I guess I cannot get the octave from ppa.
When I issue: `sudo apt-get update -qq -y` I get the following:
```
W: The repository 'http://ppa.launchpad.net/octave/stable/ubuntu xenial Release' does not have a Release file.
W: The repository 'http://ppa.launchpad.net/openjdk/ppa/ubuntu xenial Release' does not have a Release file.
E: Failed to fetch http://ppa.launchpad.net/octave/stable/ubuntu/dists/xenial/main/binary-amd64/Packages  404  Not Found
E: Failed to fetch http://ppa.launchpad.net/octave/stable/ubuntu/dists/xenial/main/binary-i386/Packages  404  Not Found
E: Failed to fetch http://ppa.launchpad.net/openjdk/ppa/ubuntu/dists/xenial/main/binary-amd64/Packages  404  Not Found
E: Failed to fetch http://ppa.launchpad.net/openjdk/ppa/ubuntu/dists/xenial/main/binary-i386/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
```
I went to the URL http://ppa.launchpad.net/openjdk/ppa/ubuntu/dists and it seems it has no packages for `xenial`, i.e. Ubuntu 16.04
.
Although `sudo apt-get install octave liboctave-dev -qq -y` installed octave-4.0.0 to /opt dir and I was able to successfully build pytave with it!!

**Comment by cbm755** [comment]
Guess you know now, but without the PPA we get Octave 3.8 (last I checked).

> Although sudo apt-get install octave liboctave-dev -qq -y installed octave-4.0.0 to /opt dir and I
> was able to successfully build pytave with it!!

You mean on your own machine right?  But then why Travis cannot build against Octave 4.0 I wonder...  Let me try building PyTave against Octave 4 on my machine...

**Comment by cbm755** [comment]
I can build mtmiller/pytave tip against Octave 4.0.1 on Fedora 24.  Hmmm, maybe Octave 4.0.0 versus 4.0.1?  @mtmiller, can you look at the Travis CI output for 34bf2cd and offer an opinion?

Here's some of the output for convenience:
````
Pytave is configured with the following setup

Dependencies

  Octave ............. /usr/include/octave-4.0.0

  Python ............. -I/opt/python/2.7.10/include/python2.7

    library .......... -L/opt/python/2.7.10/lib -lpython2.7

    executable ....... /home/travis/virtualenv/python2.7.10/bin/python

  Boost::Python ...... boost_python

Features

  uselocale .......... yes

========================================================================

make  all-am

make[1]: Entering directory `/home/travis/build/cbm755/octsympy/pytave'

/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -std=gnu++11 -DHAVE_CONFIG_H -I.  -I/usr/include/octave-4.0.0 -I/usr/include/octave-4.0.0/octave -I/opt/python/2.7.10/include/python2.7 -I/home/travis/virtualenv/python2.7.10/lib/python2.7/site-packages/numpy/core/include   -g -O2 -MT _pytave_la-pytave.lo -MD -MP -MF .deps/_pytave_la-pytave.Tpo -c -o _pytave_la-pytave.lo `test -f 'pytave.cc' || echo './'`pytave.cc

libtool: compile:  g++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I/usr/include/octave-4.0.0 -I/usr/include/octave-4.0.0/octave -I/opt/python/2.7.10/include/python2.7 -I/home/travis/virtualenv/python2.7.10/lib/python2.7/site-packages/numpy/core/include -g -O2 -MT _pytave_la-pytave.lo -MD -MP -MF .deps/_pytave_la-pytave.Tpo -c pytave.cc  -fPIC -DPIC -o .libs/_pytave_la-pytave.o

In file included from /usr/include/octave-4.0.0/octave/dim-vector.h:35:0,

                 from /usr/include/octave-4.0.0/octave/Array.h:35,

                 from /usr/include/octave-4.0.0/octave/boolMatrix.h:27,

                 from /usr/include/octave-4.0.0/octave/mx-base.h:32,

                 from /usr/include/octave-4.0.0/octave/Matrix.h:30,

                 from /usr/include/octave-4.0.0/octave/oct.h:33,

                 from pytave.cc:32:

/usr/include/octave-4.0.0/octave/oct-refcount.h:27:3: error: #error "The file <octave/config.h> must be included before oct-refcount.h."

 # error "The file <octave/config.h> must be included before oct-refcount.h."

   ^

In file included from /usr/include/octave-4.0.0/octave/mx-base.h:28:0,

                 from /usr/include/octave-4.0.0/octave/Matrix.h:30,

                 from /usr/include/octave-4.0.0/octave/oct.h:33,

                 from pytave.cc:32:

/usr/include/octave-4.0.0/octave/MatrixType.h:36:1: error: variable ‘OCTAVE_API MatrixType’ has initializer but incomplete type

 MatrixType

 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:38:1: error: expected primary-expression before ‘public’

 public:

 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:38:1: error: expected ‘}’ before ‘public’

/usr/include/octave-4.0.0/octave/MatrixType.h:38:1: error: expected ‘,’ or ‘;’ before ‘public’

/usr/include/octave-4.0.0/octave/MatrixType.h:57:20: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (void);

                    ^

/usr/include/octave-4.0.0/octave/MatrixType.h:59:21: error: ‘MatrixType’ does not name a type

   MatrixType (const MatrixType &a);

                     ^

/usr/include/octave-4.0.0/octave/MatrixType.h:59:33: error: ISO C++ forbids declaration of ‘a’ with no type [-fpermissive]

   MatrixType (const MatrixType &a);

                                 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:59:35: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const MatrixType &a);

                                   ^

/usr/include/octave-4.0.0/octave/MatrixType.h:61:31: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const Matrix &a);

                               ^

/usr/include/octave-4.0.0/octave/MatrixType.h:63:38: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const ComplexMatrix &a);

                                      ^

/usr/include/octave-4.0.0/octave/MatrixType.h:65:36: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const FloatMatrix &a);

                                    ^

/usr/include/octave-4.0.0/octave/MatrixType.h:67:43: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const FloatComplexMatrix &a);

                                           ^

/usr/include/octave-4.0.0/octave/MatrixType.h:69:37: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const SparseMatrix &a);

                                     ^

/usr/include/octave-4.0.0/octave/MatrixType.h:71:44: error: expected constructor, destructor, or type conversion before ‘;’ token

   MatrixType (const SparseComplexMatrix &a);

                                            ^

/usr/include/octave-4.0.0/octave/MatrixType.h:73:21: error: ‘matrix_type’ does not name a type

   MatrixType (const matrix_type t, bool _full = false);

                     ^

/usr/include/octave-4.0.0/octave/MatrixType.h:73:33: error: ISO C++ forbids declaration of ‘t’ with no type [-fpermissive]

   MatrixType (const matrix_type t, bool _full = false);

                                 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:73:55: error: expected constructor, destructor, or type conversion before ���;’ token

   MatrixType (const matrix_type t, bool _full = false);

                                                       ^

/usr/include/octave-4.0.0/octave/MatrixType.h:75:21: error: ‘matrix_type’ does not name a type

   MatrixType (const matrix_type t, const octave_idx_type np,

                     ^

/usr/include/octave-4.0.0/octave/MatrixType.h:75:33: error: ISO C++ forbids declaration of ‘t’ with no type [-fpermissive]

   MatrixType (const matrix_type t, const octave_idx_type np,

                                 ^

/usr/include/octave-4.0.0/octave/MatrixType.h:75:42: error: ‘octave_idx_type’ does not name a type

   MatrixType (const matrix_type t, const octave_idx_type np,

                                          ^

/usr/include/octave-4.0.0/octave/MatrixType.h:75:58: error: ISO C++ forbids declaration of ‘np’ with no type [-fpermissive]

   MatrixType (const matrix_type t, const octave_idx_type np,

                                                          ^

/usr/include/octave-4.0.0/octave/MatrixType.h:76:21: error: ‘octave_idx_type’ does not name a type

               const octave_idx_type *p, bool _full = false);

                     ^

/usr/include/octave-4.0.0/octave/MatrixType.h:76:38: error: ISO C++ forbids declaration of ‘p’ with no type [-fpermissive]

               const octave_idx_type *p, bool _full = false);

                                      ^

/usr/include/octave-4.0.0/octave/MatrixType.h:76:60: error: expected constructor, destructor, or type conversion before ‘;’ token

               const octave_idx_type *p, bool _full = false);

                                                            ^

/usr/include/octave-4.0.0/octave/MatrixType.h:78:21: error: ‘matrix_type’ does not name a type

   MatrixType (const matrix_type t, const octave_idx_type ku,

                     ^
````

**Comment by genuinelucifer** [comment]
> Guess you know now, but without the PPA we get Octave 3.8 (last I
checked).
>

Hmm. And with 3.8 I can't even seem to get doctest installed! So had to
revert it.

> You mean on your own machine right? But then why Travis cannot build
against Octave 4.0 I wonder... Let me try building PyTave against Octave 4
on my machine...
>

Yup. And 'sudo apt-get install octave' installed 4.0.0 only! But I had to
update my linux-image... Else it showed some conflict in packages.
But after updating everything else went fine.

**Comment by cbm755** [comment]
Eh, so if you can compile against 4.0.0 on your Ubuntu 16.04 machine, then why can't Travis...?

14.04 versus 16.04...?  Strange.

**Comment by genuinelucifer** [comment]
> Eh, so if you can compile against 4.0.0 on your Ubuntu 16.04 machine, then why can't Travis...?
> 14.04 versus 16.04...? Strange.

Hmmm... Probably some other library/gcc version problem?
I'll try to update gcc and tests again..

**Comment by genuinelucifer** [comment]
Finally successfully built pytave on travis! :dancer: 
.
In my repo, I just added `#include<octave/config.h?` to every `.cc` file...
Currently there are many warnings about redefined symbols but I don't have a better solution yet.
It seems that either `-DHAVE_CONFIG_H` to `g++` is not defining `HAVE_CONFIG_H` properly OR `#include<config.h>` is not working as expected...
I'll try to run octave with pytave next...

**Comment by genuinelucifer** [comment]
Strange! It is getting exception when doing `syms x` at the line `if ((exist('pyeval') > 1) && (exist('pyexec') > 1))`!!

**Comment by genuinelucifer** [comment]
As per http://savannah.gnu.org/bugs/?36067, it seems that the oct files are causing seg fault!
.
don't know what went wrong now. Most probably due to buggy octave-4.0.0! Or maybe due to redefined symbols!
Since octave-4.0.2 is already out, when can it be expected with `sudo apt-get install octave`?!

**Comment by mtmiller** [comment]
There's a lot going on here, I'll try to lay out what I know and see where that leaves you.

* The PPA `ppa:octave/stable` is maintained by me, and it is intended to provide a backport of whatever version of Octave is in the latest development Ubuntu release. I'm currently a little behind on updating to 4.0.2, so it is still 4.0.0.

* There are other PPAs providing builds of the development branch of Octave, for example `ppa:dac922/octave-unstable` [here](https://launchpad.net/~dac922/+archive/ubuntu/octave-unstable).

* If you are getting segfaults from simply loading oct-file functions, that may mean they were built for the wrong version of Octave. Make sure you are actually building against and running the oct-files in the same version of Octave.

* Octave's API and include files have changed _drastically_ from version 4.0.x to the development version (what will become 4.2.x). I made changes back in April to support the new API and not even attempt to stay compatible with the 4.0.x API, see [386772f4e12d](https://bitbucket.org/mtmiller/pytave/commits/386772f4e12ddcfc814caa758d74effc98c2cacd). The errors you quoted seem to be exactly the kind of problems that those header files hacks were in place to work around.

So in short I would recommend always building and testing with pytave using Octave's development branch. I guess that using a PPA that follows Octave's development branch would be the best approach.

**Comment by genuinelucifer** [comment]
Thanks @mtmiller, I have triggered a build with dev branch of octave (from source).. I don't know how long it will take, but I hope it's ok to build it on travis (atleast just for testing 1-2 times, untill we can get it from ppa)...

**Comment by cbm755** [comment]
Don't build octave from source on Travis: it'll just warm up the fucking planet.  @mtmiller mentions dev PPAs above.

**Comment by cbm755** [comment]
oh I see just for testing, well ok then :)

**Comment by genuinelucifer** [comment]
On Jun 20, 2016 11:48 PM, "Colin Macdonald" <notifications@github.com>
wrote:
>
> Don't build octave from source on Travis: it'll just warm up the fucking
planet. @mtmiller mentions dev PPAs above.
>

Oh sorry.
I can't stop it though (no option). Could you pls stop it.

**Comment by cbm755** [comment]
I could, but let's just let it go, maybe helps you debug something.

**Comment by genuinelucifer** [comment]
Well, it exceeded max time allowed by travis. So, we can't possibly build octave on travis. We'll have to wait for the ppa update then.
.
Also, how does  'sudo apt-get install octave' works? I mean who updates the octave that is downloaded via this command?

**Comment by genuinelucifer** [comment]
I meant without adding the ppa.

**Comment by mtmiller** [comment]
I think you're getting a little distracted. I already gave you a pointer to a PPA that will install a development version of Octave. It is updated by another volunteer, it may not be the absolute latest off the default branch, but you can ping him and he can probably update it easily enough.

I am in the process of update the `stable` PPA this afternoon, so that will provide 4.0.2 on all Ubuntu LTS releases.

The primary Ubuntu repository is maintained by Ubuntu packagers, and will probably never be updated beyond 4.0.0 for Ubuntu 16.04, because that's the way LTS works. The 16.10 release will have Octave 4.0.2.

**Comment by cbm755** [comment]
Let's ignore this for a bit then.  `pytave_ex` needs work: after all, no point in having CI if we have known failing tests!

**Comment by genuinelucifer** [comment]
If I build pytave from `mtmiller/pytave` then travis shows the exactly same errors as it was for octave-4.0.0!
And  similarly, if I build from my fork then it gives SEGFAULT in the same statement as before!

**Comment by cbm755** [comment]
I'm not sure if this is the problem but there is a strange mix of libraries in there:

1. Python from /opt/python/2.7.10/
2. /usr/include/boost/python/
3. /home/travis/virtualenv/python2.7.10/bin/python

I would guess that boost is linked against the Ubuntu system python... potential for trouble.

Let me reiterate that the `pytave_ex` branch is more important than this.

**Comment by genuinelucifer** [comment]
Travis is not running, so it's difficult to tell, but I hope we can see on github that (2 out of 3 checks passed)!
Is the setting ok now Or should I remove the `PYTAVE=yes` env?

**Comment by cbm755** [comment]
that's because I had errors in the .travis file---they have a lint checker.

see `ci_pytave2` again

**Comment by cbm755** [comment]
I suspect your change won't work because I changed `language` to generic: but give it a try...

probably you don't want `- python: 2.7` on the line that has `PYTAVE = yes`: I think it will build a virtualenv which was exactly the linking problem I worked around with `language =generic`.

good luck!

**Comment by genuinelucifer** [comment]
I guess It works, as expected, now. 2/3 builds pass. Pytave one fails with a few of the tests failing...
Although I am not ssure why it shows "All checks have failed" on github!
.
But, when the `pytave_ex` will be merged then the number of failing tests will decrease...
.
> probably you don't want - python: 2.7 on the line that has PYTAVE = yes

Probably it works even with that. Should I remove it?
.
Also, Is this merge-able now or should we try to build it with `mtmiller/pytave`?

**Comment by cbm755** [comment]
Maybe prepare a new PR against `pytave` branch with all this squashed, then I'll review.

Do you work in that repo?  We may need a different fork, just for use with travis...

**Comment by genuinelucifer** [comment]
> Maybe prepare a new PR against pytave branch with all this squashed, then
I'll review.
>

Sure.

> Do you work in that repo? We may need a different fork, just for use with
travis...
>

Yes I do use that fork. I could make a new fork though, if you want it to
be for Travis. Or we can make a new fork for Travis.
